### PR TITLE
[DomCrawler] always pass base href to subcrawlers

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

Make sure that all relevant information is passed to created crawlers.
To avoid future regressions, this commit backports the approach taken by
@stof in #15934 to have a single place in the class that is responsible
to create subcrawler instances.

### CODE REVIEW

File: src/Symfony/Component/DomCrawler/Crawler.php [hunk 1]
```diff
@@ -370,7 +370,7 @@ public function each(\Closure $closure)
      */
     public function slice($offset = 0, $length = -1)
     {
-        return new static(iterator_to_array(new \LimitIterator($this, $offset, $length)), $this->uri);
```
Comment (stof) [inline src/Symfony/Component/DomCrawler/Crawler.php]: for reference, this is the place where the baseHref was missing.

**Comment by stof** [comment]
I'm :+1: on backporting this

File: src/Symfony/Component/DomCrawler/Crawler.php (hunk 1 above)
Comment (xabbuh) [inline src/Symfony/Component/DomCrawler/Crawler.php]: Yes, in #15934 I asked to backport this to the `2.3` branch, but realised that the `slice()` method was introduced later. That's why I did not backport your changes to Symfony 2.3.

**Comment by Tobion** [comment]
:+1: 

Status: Reviewed

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**